src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -135,6 +135,14 @@ class ClipBam
       out += rec
     }
 
+    // Set the "All" and "Pair" read types
+    metricsMap.foreach { metrics: Map[ReadType, ClippingMetrics] =>
+      val typesForPair: Set[ReadType] = Set(ReadType.ReadOne, ReadType.ReadTwo)
+      val typesForAll: Set[ReadType]  = typesForPair ++ Set(ReadType.Fragment)
+      metrics(ReadType.Pair).add(metrics.filter { case (tpe, _) => typesForPair.contains(tpe) }.values.toSeq:_*)
+      metrics(ReadType.All).add(metrics.filter { case (tpe, _) => typesForAll.contains(tpe) }.values.toSeq:_*)
+    }
```

Comment:
I found this block a little hard to follow.  It may be personal preference, but I offer the below alternative for your consideration:

```scala
import ReadType._
metricsMap.foreach { 
    case (Fragment, metric) => metricMap(All).add(metric)
    case (ReadOne | ReadTwo, metric) => Seq(Pair, All).foreach { r => metricMap(r).add(metric) }
    case _ => Unit
}